### PR TITLE
Implements #3069 - allows use of pre-hashed passwords with userpass backend

### PR DESCRIPTION
**Pull request by Ivan-Roger**
This is a cherry-pick of changes suggested in hashicorp/vault#3225.

### CODE REVIEW

File: website/content/api-docs/auth/userpass.mdx [hunk 1]
```diff
@@ -27,8 +27,8 @@ Create a new user or update an existing user. This path honors the distinction b
 ### Parameters
 
 - `username` `(string: <required>)` – The username for the user. Accepted characters: alphanumeric plus "_", "-", "." (underscore, hyphen and period); username cannot begin with a hyphen, nor can it begin or end with a period.
-- `password` `(string: <required>)` - The password for the user. Only required
-  when creating the user.
+- `password` `(string: <required if password_hash is not given>)` - The password for the user. Only required when creating the user.
+- `password_hash` `(string: <required if password is not given>)` - Pre-hashed version of this user's password in bcrypt format. Mutually exclusive with `password`.
```
Comment (schavis) [inline website/content/api-docs/auth/userpass.mdx:31]:
```suggestion
- `password_hash` `(string: <required if password is not given>)` - Pre-hashed password for the current user in bcrypt format. Mutually exclusive with `password`.
```

File: website/content/api-docs/auth/userpass.mdx [hunk 2]
```diff
@@ -27,8 +27,8 @@ Create a new user or update an existing user. This path honors the distinction b
 ### Parameters
 
 - `username` `(string: <required>)` – The username for the user. Accepted characters: alphanumeric plus "_", "-", "." (underscore, hyphen and period); username cannot begin with a hyphen, nor can it begin or end with a period.
-- `password` `(string: <required>)` - The password for the user. Only required
-  when creating the user.
+- `password` `(string: <required if password_hash is not given>)` - The password for the user. Only required when creating the user.
```
Comment (schavis) [inline website/content/api-docs/auth/userpass.mdx:30]:
```suggestion
- `password` `(string: <required if password_hash is not given>)` - Password for the current user. Only required when creating the user.  Mutually exclusive with `password_hash`.
```

updated to match the language in `password_hash`

**Review by schavis — CHANGES_REQUESTED** [review]

**Comment by schavis** [comment]
Content changes lgtm. Please make sure to sign the CLA so your changes can be merged once someone from the Vault ENG team signs off on the code changes

**Review by schavis — APPROVED** [review]

**Comment by Ivan-Roger** [comment]
@schavis Unfortunately the missing CLA signature comes from the original author (I cherry-picked their commit). He does not seem to be active anymore. Would it be possible to bypass his signature ?

**Comment by peteski22** [comment]
Hi @Ivan-Roger thanks for trying to get this over the line, as the original contributor hasn't signed the CLA, I'm going to move this on via a different method.

I will close your PR in favour of https://github.com/hashicorp/vault/pull/25862

**Pull request closed without merging**